src/pybind/mgr/cephadm/inventory.py

```diff
@@ -217,6 +224,19 @@ def load(self):
                     host, e))
                 pass
 
+    def update_osdspec_previews(self, search_host: str = ''):
+        # Set global 'pending' flag for host
+        self.loading_osdspec_preview[search_host] = True
+        previews = []
+        # query OSDSpecs for host <search host> and generate/get the preview
+        for preview in self.mgr.osd_service.get_previews(search_host):
+            # There can be multiple previews for one host due to multiple OSDSpecs.
+            previews.append(preview)
+        self.mgr.log.debug(f"Loading OSDSpec previews to HostCache")
+        self.osdspec_previews[search_host] = previews
+        # Unset global 'pending' flag for host
+        self.loading_osdspec_preview[search_host] = False
+
```

Comment:
Hm would it make sense to move this into the OSDService class? The other methods in here won't do any remote calls.

Reply:
I put it there to have similar logic in one place. For a scope standpoint it would probably be best suited in OSDService().. 
Since this is mostly cache related and we just source the to-be-cached data from OSDService, I would keep it here.

Reply:
hm. It's indeed cache related, but it's the first thing in this module to do remote calls.